### PR TITLE
i18n(fr): add `guides/overriding-components.md` & `reference/overrides.md`

### PR DESCRIPTION
**Pull request by HiDeoo**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

This pull request adds the french translation of the `guides/overriding-components.md` and `reference/overrides.md` pages (including the fix from #833).

Regarding the translation of `override` in french, there are multiple possibilities:

- `remplacer` which would be the most accurate translation for the common `replace` verb.
- `surcharger` which would be the most accurate translation for `overload` in the context of programming, e.g. Java method overloading.
- `redéfinir` which would be the most accurate translation for `override` in the context of programming, e.g. Java method overriding.

Considering the definitions of `overloading` and `overriding` in the context of programming, I went with `redéfinir` (verb) and `redéfinition` (noun) for the translation of `override` in the context of Starlight components.

A small note about `render` too which is used a lot in these pages, as far as I know, we do not really use the translation of the verb `render` in french in this context. This is also visible in the [french React docs](https://fr.react.dev/reference/react-dom/server/renderToString) where `rendre` is never used compared to the [english version](https://react.dev/reference/react-dom/server/renderToString). We have the word `rendu` that can be used but it is a noun, so in this translation, when a verb is needed I often fallback to `utilisé` (used) or `affiché` (displayed).

_I am also taking part in Hacktoberfest._

### CODE REVIEW

File: docs/src/content/docs/fr/reference/overrides.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+tableOfContents:
+  maxHeadingLevel: 4
+---
+
+Vous pouvez redéfinir les composants intégrés à Starlight en spécifiant des chemins vers des composants de remplacement avec l'option de configuration [`components`](/fr/reference/configuration#components) de Starlight.
+Cette page répertorie tous les composants disponibles qui peuvent être redéfinis et fournit des liens vers leurs implémentations par défaut sur GitHub.
+
+Pour en savoir plus, consultez le [guide des redéfinitions de composants](/fr/guides/overriding-components/).
+
+## Props des composants
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/overrides.md:13]: Using "props" like the [Astro docs](https://docs.astro.build/fr/core-concepts/astro-components/#props-de-composant) for consistency.

File: docs/src/content/docs/fr/reference/overrides.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+**Type:** `string`
+
+L’étiquette d’identification BCP-47 pour la langue de la page, par exemple `en`, `zh` ou `pt-BR`.
+
+#### `locale`
+
+**Type:** `string | undefined`
+
+Le chemin de base utilisé pour servir une langue. `undefined` pour les slugs de la locale racine.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/overrides.md:50]: As far as I know, there is no french word for "slug" in this context. Also [not translated](https://developer.mozilla.org/fr/docs/Glossary/Slug) in the french version of MDN.

File: docs/src/content/docs/fr/reference/overrides.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+**Type:** `URL | undefined`
+
+Un objet `URL` de l'adresse où cette page peut être modifiée si cette fonctionnalité est activée.
+
+---
+
+## Composants
+
+### Métadonnées
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/overrides.md:141]: "head" and "header" are usually translated both using "en-tête" which would be annoying in this page to use the same word so I used "Métadonnées" like on [MDN](https://developer.mozilla.org/fr/docs/Web/HTML/Element/head).

File: docs/src/content/docs/fr/reference/overrides.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+Ces composants sont responsables de la mise en page des composants de Starlight et de la gestion des vues pour différents points d'arrêt.
+Redéfinir ceux-ci implique une complexité significative.
+Lorsque cela est possible, préférez redéfinir un composant de plus bas niveau.
+
+#### `PageFrame`
+
+**Composant par défaut:** [`PageFrame.astro`](https://github.com/withastro/starlight/blob/main/packages/starlight/components/PageFrame.astro)
+
+Composant de mise en page contenant la plupart du contenu de la page.
+L'implémentation par défaut configure la mise en page de l'en-tête, de la barre latérale et du contenu principal et inclut des emplacements (slots) nommés `header` et `sidebar` en plus de l'emplacement par défaut pour le contenu principal.
```
Comment (HiDeoo) [inline docs/src/content/docs/fr/reference/overrides.md:187]: Using the same syntax as the [Astro Docs](https://docs.astro.build/fr/core-concepts/astro-components/#les-emplacements-slots-nomm%C3%A9s).

File: docs/src/content/docs/fr/guides/overriding-components.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+</style>
+```
+
+Pour en savoir plus sur les props disponibles, consultez la [référence des redéfinitions](/fr/reference/overrides/#props-des-composants).
+
+### Redéfinir uniquement sur des pages spécifiques
+
+Les redéfinitions de composants s'appliquent à toutes les pages. Cependant, vous pouvez les utiliser conditionnellement grâce aux valeurs de `Astro.props` pour déterminer quand afficher votre interface utilisateur personnalisée, quand afficher l'interface utilisateur par défaut de Starlight, ou même quand afficher quelque chose de complètement différent.
+
+Dans l'exemple suivant, un composant redéfinissant le composant [`Footer`](/fr/reference/overrides/#footer-1) de Starlight affiche "Construit avec Starlight 🌟" sur la page d'accueil uniquement, et affiche sinon le pied de page par défaut sur toutes les autres pages :
```
Comment (delucis) [inline docs/src/content/docs/fr/guides/overriding-components.md:111]: This link needs to be `#footer` — in English we have two identical headings — “Footer” and “`Footer`” — but in French these are “Pied de page” and “`Footer`”, so this heading doesn’t have the extra number added.

```suggestion
Dans l'exemple suivant, un composant redéfinissant le composant [`Footer`](/fr/reference/overrides/#footer) de Starlight affiche "Construit avec Starlight 🌟" sur la page d'accueil uniquement, et affiche sinon le pied de page par défaut sur toutes les autres pages :
```

**Review by delucis — APPROVED** [review]
Looks great — thanks @HiDeoo! Just noticed one link issue that also came up in another translation PR.

**Comment by HiDeoo** [comment]
Add myself the `hacktoberfest-accepted` tag before merging, hope that is ok.

File: docs/src/content/docs/fr/guides/overriding-components.md (hunk 5 above)
Comment (HiDeoo) [reply docs/src/content/docs/fr/guides/overriding-components.md]: Damn, excellent catch, thanks a lot for the review 🙌

**Pull request merged**